### PR TITLE
fix admin v2 entrance editing and redirect

### PR DESCRIPTION
**Pull request by denisdimitrov**
- Admin v2 redirection wasn't working when on save
- Fix bug where map background was being deleted after save
- Fix typo
- Pass prop to open inputs when already populated

Fixes: https://github.com/sparkletown/internal-sparkle-issues/issues/237

### CODE REVIEW

File: functions/venue.js [hunk 1]
```diff
@@ -648,8 +648,6 @@ exports.updateVenue_v2 = functions.https.onCall(async (data, context) => {
 
   if (data.mapBackgroundImageUrl) {
     updated.mapBackgroundImageUrl = data.mapBackgroundImageUrl;
-  } else {
-    updated.mapBackgroundImageUrl = "";
   }
```
Comment (0xdevalias) [inline functions/venue.js:651]: Do we have any features currently that want to be able to delete the `mapBackgroundImageUrl`? Or will it always just be replaced with a new url; never deleted/set back to default?

File: src/pages/Admin/EntranceExperience/EntranceExperience.tsx [hunk 2]
```diff
@@ -128,6 +128,7 @@ const EntranceExperience: React.FC<EntranceExperienceProps> = ({
               register={register}
               fieldName="entrance"
               showTitle={false}
+              editing={venue.entrance}
```
Comment (0xdevalias) [inline src/pages/Admin/EntranceExperience/EntranceExperience.tsx:131]: Should this also have an `errors` key set like the other examples here?

**Review by 0xdevalias — APPROVED** [review]
[CRO] Tentatively approved, assuming satisfactory answers to the questions raised in https://github.com/sparkletown/sparkle/pull/1107#pullrequestreview-577982976

File: functions/venue.js (hunk 1 above)
Comment (denisdimitrov) [reply functions/venue.js:651]: We do not have any features that delete the mapBackgroundImageUrl. This update functions updates other venue information and should always persist the background image.

File: src/pages/Admin/EntranceExperience/EntranceExperience.tsx (hunk 2 above)
Comment (denisdimitrov) [reply src/pages/Admin/EntranceExperience/EntranceExperience.tsx:131]: Good point. I'll add those too.

File: functions/venue.js (hunk 1 above)
Comment (0xdevalias) [reply functions/venue.js:651]: Awesome, thanks for the context.

**Review by mike-lvov — APPROVED** [review]
LGTM

**Comment by denisdimitrov** [comment]
@sparkletown/engineering The CI fails saying that prettier needs to be run. But when I run it locally it says this: `All matched files use Prettier code style!` Any ideas what the problem might be?

**Comment by 0xdevalias** [comment]
> The CI fails saying that prettier needs to be run. But when I run it locally it says this: `All matched files use Prettier code style!` Any ideas what the problem might be?

@denisdimitrov Have you ‘npm install’d on this branch since prettier got updated?

File: src/pages/Admin/EntranceExperience/EntranceExperience.tsx [hunk 3]
```diff
@@ -84,7 +87,9 @@ const EntranceExperience: React.FC<EntranceExperienceProps> = ({
     },
     [onSave, user, venue.name]
   );
+  console.log(venue.entrance);
 
+  console.log(getValues());
```
Comment (0xdevalias) [inline src/pages/Admin/EntranceExperience/EntranceExperience.tsx:92]: These should be removed

File: src/pages/Admin/EntranceExperience/EntranceExperience.tsx [hunk 4]
```diff
@@ -128,13 +133,13 @@ const EntranceExperience: React.FC<EntranceExperienceProps> = ({
...
+              editing={venue.entrance}
+              errors={errors.entrance}
             />
           </S.ItemBody>
         </S.ItemWrapper>
 
-        <Button type="submit" disabled={!dirty}>
-          Save
-        </Button>
+        <Button type="submit">Save</Button>
```
Comment (0xdevalias) [inline src/pages/Admin/EntranceExperience/EntranceExperience.tsx:142]: Why did we remove the `disabled` from this?

File: src/pages/Admin/Venue/EntranceInput.tsx [hunk 5]
```diff
@@ -101,6 +114,9 @@ const EntranceInput: React.FC<EntranceInputProps> = ({
           />
           <Form.Label>Video URL</Form.Label>
           <Form.Control ref={register} name={videoUrl} custom />
+          {errors && errors[index].videoUrl && (
```
Comment (0xdevalias) [inline src/pages/Admin/Venue/EntranceInput.tsx:117]:
```suggestion
          {errors?.[index]?.videoUrl && (
```

**Review by 0xdevalias — CHANGES_REQUESTED** [review]

File: src/pages/Admin/EntranceExperience/EntranceExperience.tsx (hunk 4 above)
Comment (denisdimitrov) [reply src/pages/Admin/EntranceExperience/EntranceExperience.tsx:142]: The fields have validation already. And the `dirty` state is not changing to true, after calling `clear` or `remove` from `useForm`

**Comment by denisdimitrov** [comment]
@0xdevalias Comments have been fixed.

File: src/pages/Admin/EntranceExperience/EntranceExperience.tsx (hunk 4 above)
Comment (0xdevalias) [reply src/pages/Admin/EntranceExperience/EntranceExperience.tsx:138]: Makes sense.

I'm not sure that our other forms have the 'disable the submit button' sort of logic on them anyway do they? If there is a pattern that we already have that handles this well that you can copy from, that would be cool. But if this was a one off/different from our standards, then no need to try and implement it.

**Comment by 0xdevalias** [comment]
@denisdimitrov Do you know why these commits are showing up as part of this PR? It's making the code diff hard to review since it's including unrelated things.

image

**Review by 0xdevalias — APPROVED** [review]

File: src/pages/Admin/EntranceExperience/EntranceExperience.tsx (hunk 4 above)
Comment (denisdimitrov) [reply src/pages/Admin/EntranceExperience/EntranceExperience.tsx:138]: Yeah, this disable pattern is new and should be removed imo, just to keep it consistent with the other ux flow.

**Comment by denisdimitrov** [comment]
> Do you know why these commits are showing up as part of this PR? It's making the code diff hard to review since it's including unrelated things.

Yes, I accidentally rebased with staging instead of this PR. My bad.

File: src/pages/Admin/EntranceExperience/EntranceExperience.tsx (hunk 4 above)
Comment (0xdevalias) [reply src/pages/Admin/EntranceExperience/EntranceExperience.tsx:138]: Sounds good to me

**Comment by 0xdevalias** [comment]
> Yes, I accidentally rebased with staging instead of this PR. My bad.

@denisdimitrov Cool, makes sense. Just re-skimmed through the diff now that you've fixed it and it all looks good 👍🏻

**Pull request merged**